### ISSUE TITLE
Set up tool timeout and consider no issues reported

### ISSUE THREAD

**Issue by feliam** [EthereumAnalysisBenchmarks/evm-analyzer-bench-suites#34]
Note that when manticore finishes it is because all the possible executions were explored (modulo bugs in manticore). So the confidence in a positive or negative result is high. 
At the end it also generates example testcases to reproduce each exact path it explored. So you can take this a go try it on the blockchain. .. I guess, this has nothing to do with the current benchmark. But it eats a lot of cpu time anyway. 

Proposal.
The same way you'll call it a False Negative If a fuzzer running for fixed period of time fails to find the known bug in some program. I propose you add mcore a  `--timeout 4` and interpret it's output as is (no findings if no findings).

**Comment by rocky** [comment]
I am not sure I understand: you want another "don't know" or "no findings" category? 

I suppose okay. Does manticore report this on its own right now in a simple explicit way?

 If you or someone else implement it, I suppose we can add it, especially if people upvote that issue.

**Comment by rocky** [comment]
I should note that there is a [`run_time` field](https://github.com/EthereumAnalysisBenchmarks/evm-analyzer-bench-suites/blob/master/benchconf/Suhabe-Manticore.yaml#L14-L19) that you can add for a particular benchmark to suggest that you shouldn't try to run the benchmark unless you are willing to wait longer that _n_ seconds.

Yes, this is different.

**Comment by b-mueller** [comment]
If I understand correctly, Manticore will run until all paths are explored by default so we need to set a reasonable timeout. @feliam is the timeout specified in seconds?

**Comment by feliam** [comment]
Yes I meant what prof. b-mueller clarified. timeout -> nofindings  

I think you add manticore here to be able to compare awesome mythril with something. Though AFAIK both tools do similar but different things. mcore is spending a lot of resources to make reproducible testcases for each state. That time is sunk and do not really add to the comparation. Mcore will normally never finish in any real case in a 5 minutes timebox. In a few days running maybe.

Or am I confused and mythril also generates the full reproducible tx trace and mcore is doing something terribly wrong ?

**Comment by b-mueller** [comment]
So mcore cannot be used for "point-and-shoot" analysis a'la Mythril? Tbh I'm not too familiar with it, but I was assuming that it does detect most vulnerability classes generically. If that's not the case, then it really might not make sense to include it in the benchmark.

> mcore is spending a lot of resources to make reproducible testcases for each state. That time is sunk and do not really add to the comparation.

By "each state", do you refer to each state that possible constitutes a vulnerability, or for every state that is reached? Mythril only uses the solver when it suspects a vulnerability.

> Or am I confused and mythril also generates the full reproducible tx trace and mcore is doing something terribly wrong ?

For the reported issues, we can also compute data that could be used to reproducible test cases. It would only be a matter of assembling transactions from the solver outputs to transactions. However we haven't implemented that yet.

I think it's worth repeating that purpose of these benchmarks is to provide concrete examples of standard vulnerability classes and not to make a particular tool look good. We're trying to make sure of that by also including other benchmarks than  ours.If Manticore is not intended to be used to as a generic vulnerability detector (or at least not by running it in a sub-10-minutes timespan), then we should probably not include it in the benchmarks, which is fine.

**Comment by b-mueller** [comment]
> prof. b-mueller

Also, I'm no professor if that's what you meant to say. Just regular nerd 🤓
